USAGE_IN_PROJECT.md

```diff
@@ -0,0 +1,185 @@
+doesn't, you'll need to manually install the extension another way.
+
+**Steps**
+
+1. Open the cloned directory in VSCode
+
+2. Launch a Dev Container by clicking the pop-up prompt in the lower right of
+the VS Code window that suggests to "Reopen in Container"
+
+   ![Shows "Reopen in Container" prompt when opening the cccl directory in VScode.](./docs-img/reopen_in_container.png)
+
+   - If the pop-up prompt doesn't come up, use the Command Palette to start a Dev Container. Press `[Ctrl|CMD]+Shift+P` to open the Command Palette. Type "Remote-Containers: Reopen in Container" and select it.
+
+     ![Shows "Reopen in Container" in command pallete.](./docs-img/open_in_container_manual.png)
+
+3. Select an environment with the desired build tools from the list:
+
+   ![Shows list of available container environments.](./docs-img/container_list.png)
+
+   The available tools depend on your project. These can be configured when
+   generating the matrix of base images. See [the docs on adapting devcontainers](./USAGE.md#custom-devcontainers)
+   for more info.
+
+4. VSCode will initialize the selected Dev Container. This can take a few
+minutes the first time, as it downloads the base docker image and runs any
+specified feature installations. These steps are cached, so subsequent runs are
+faster.
+
+5. Once initialized, the local project directory is mounted into the container
+to ensure any changes are persistent.
+
+6. You project should now look like any other VS Code project, except that the
+blue box in the lower left of the VS Code window should now read `Dev Container
+@ <hostname>`. You're done! Any terminal you open will have the build
+environment activated appropriately.
+
+7. The devcontainers add build scripts that fit general usage. Type `build-` and hit
+`TAB` to see options for your project. Check the contributing guide in your repo
+for further instructions.
+
+**Exiting the devcontainer**
+
+If you are in a VS Code devcontainer on a remote (SSH) machine, you can run
+`CTRL|CMD + SHIFT + P` and select `Dev Containers: Reopen in SSH` to return to
+your host machine.
+
+### Docker (Manual Approach)
+
+Your project may have its own launch scripts that account for options in
+libraries and/or tools. The steps here should work with any repo that uses
+devcontainers, but any repo-specific scripts and instructions will probably work
+better.
+
+**Prerequisites**
+
+- [Devcontainer CLI ](https://github.com/devcontainers/cli) - needs NodeJS/npm
+
+**Steps**
+
+1. Download the [launch-devcontainer.sh script](./scripts/launch-devcontainer.sh) and
+  put it somewhere on PATH. If your project has its own launch script, use it
+  here instead.
+
+2. Set your current working directory to the root of your repo containing the
+.devcontainer folder
+
+3. Run the launch-container.sh script. Called without arguments, you'll get a menu of containers to choose from:
+
+```
+$ ./launch-devcontainer.sh
```

Comment:
If you put `launch-devcontainer.sh` on your `PATH`, do you need the `./` here?
```suggestion
$ launch-devcontainer.sh
```

Reply:
And does it have to be in your path to work?

Reply:
No, but you have to call it from a working directory inside a git repo whose devcontainer you want to launch. For example, you can call this from anywhere inside a cudf repository, but you’d have to call the launch script via its full (absolute or relative) path. The script is really designed to live in your PATH for this reason.

Reply:
I don't want to explain how to put something on PATH, but I think Brad's description here of why this script probably should be on PATH makes a lot of sense to include. I'll write something up.

Reply:
If we put this script in a `bin`  and removed the .sh suffix, that might help indicate the intent.